### PR TITLE
Made placeBlock easier

### PR DESCRIPTION
**Pull request by SMEDjs**
added option to use face `top`, `bottom`, `east`, `west`, `south` or `north`  or vector

### CODE REVIEW

File: lib/plugins/place_block.js [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
-  async function placeBlockWithOptions (referenceBlock, faceVector, options) {
+  async function placeBlockWithOptions (referenceBlock, face, options) {
+    let faceVector = face
+    if (typeof face === 'string') {
+      const directions = {
+        top: new Vec3(0, 1, 0),
+        bottom: new Vec3(0, -1, 0),
+        east: new Vec3(1, 0, 0),
+        west: new Vec3(-1, 0, 0),
+        south: new Vec3(0, 0, 1),
```
Comment (rom1504) [inline lib/plugins/place_block.js:15]: Isn't this reversed

Comment (SMEDjs) [reply lib/plugins/place_block.js:15]: it's the direction that the block face is facing at

**Comment by sefirosweb** [comment]
@SMEDjs with your PR can't be use vec3 directions directly? need to be used facevector?  
I think there should be backwards compatibility, many of us use some formula to detect the "best" position to place the block that is not based on its coordinate orientation, if there is simply an adjacent block that helps to place the block

**Comment by SMEDjs** [comment]
> @SMEDjs with your PR can't be use vec3 directions directly? need to be used facevector? I think there should be backwards compatibility, many of us use some formula to detect the "best" position to place the block that is not based on its coordinate orientation, if there is simply an adjacent block that helps to place the block

you can, it checks if it's a string but you can still use vector + if you give me the formula i can add it to the PR

**Comment by rom1504** [comment]
can you use this in a test or example ?

**Comment by rom1504** [comment]
please re-open if you want to finish it

**Pull request closed without merging**